scripts/prebuild/mdx-transforms/exclude-vault-content/index.test.mjs

```diff
@@ -0,0 +1,271 @@
+		}).rejects.toThrow(
+			'Invalid version format in directive: Vault:>=v1.invalid. Expected format: vX.Y.x',
+		)
+	})
+
+	it('should throw an error for invalid comparator', async () => {
+		const markdown = `
+<!-- BEGIN: Vault:!v1.20.x -->
+This content should throw an error.
+<!-- END: Vault:!v1.20.x -->
+`
+
+		await expect(async () => {
+			return await runTransform(markdown, vaultVersion, filePath)
+		}).rejects.toThrow(/Invalid directive format: Vault:!v1.20.x/)
+	})
+
+	it('should handle multiple version blocks correctly', async () => {
+		const markdown = `
+<!-- BEGIN: Vault:>=v1.21.x -->
+This should be removed.
+<!-- END: Vault:>=v1.21.x -->
+<!-- BEGIN: Vault:<=v1.21.x -->
+This should stay.
+<!-- END: Vault:<=v1.21.x -->
+Final content.
+`
+		const result = await runTransform(markdown, vaultVersion, filePath)
+
+		expect(result.trim()).toBe(`<!-- BEGIN: Vault:<=v1.21.x -->
+
+This should stay.
+
+<!-- END: Vault:<=v1.21.x -->
+
+Final content.`)
+	})
+
+	it('should ignore Terraform product directives and leave them untouched', async () => {
+		const markdown = `
+<!-- BEGIN: TFC:only -->
+This TFC content should be ignored by Vault transform.
+<!-- END: TFC:only -->
+<!-- BEGIN: TFEnterprise:only -->
+This TFEnterprise content should also be ignored.
+<!-- END: TFEnterprise:only -->
+<!-- BEGIN: Vault:>=v1.21.x -->
+This Vault content should be removed.
+<!-- END: Vault:>=v1.21.x -->
+Regular content that stays.
+`
+		const result = await runTransform(markdown, vaultVersion, filePath)
+
+		expect(result.trim()).toBe(`<!-- BEGIN: TFC:only -->
+
+This TFC content should be ignored by Vault transform.
+
+<!-- END: TFC:only -->
+
+<!-- BEGIN: TFEnterprise:only -->
+
+This TFEnterprise content should also be ignored.
+
+<!-- END: TFEnterprise:only -->
+
+Regular content that stays.`)
+	})
+
+	it('should throw an error for directives with products not in directiveProducts array', async () => {
+		// Override DIRECTIVE_PRODUCTS for this test only
+		vi.doMock('../build-mdx-transforms.mjs', async (importOriginal) => {
+			const actual = await importOriginal()
+			return {
+				...actual,
+				DIRECTIVE_PRODUCTS: ['Vault', 'TFC'], // CONSUL not included
+			}
+		})
+
+		// Need to re-import the transform after mocking
+		const { transformExcludeVaultContent: mockTransform } = await import(
+			'./index.mjs'
+		)
+
+		const markdown = `
+<!-- BEGIN: CONSUL:>=v1.15.x -->
+This should cause an error - CONSUL not in directiveProducts.
+<!-- END: CONSUL:>=v1.15.x -->
+`
+		const customRunTransform = async (markdown, version, filePath) => {
+			const processor = await remark()
+				.use(remarkMdx)
+				.use(mockTransform, { filePath, version })
+				.process(markdown)
+			return processor.contents
+		}
+
+		await expect(async () => {
+			return await customRunTransform(markdown, vaultVersion, filePath)
+		}).rejects.toThrow(
+			'Directive block CONSUL:>=v1.15.x could not be parsed between lines 2 and 4',
```

Comment:
I wonder if it might be better to do a regex match against the error thrown rather than an exact string match. Just to make the test more robust. That way if the error message changes for casing or punctuation, it won't fail.